### PR TITLE
[firebase_messaging] Add support for handling messages in background for iOS

### PR DESCRIPTION
**Pull request by genert**
## Description

When Flutter app is in the background or terminated allow it to handle incoming FCM messages. When the application starts an additional background channel is started to handle incoming messages when the app is not in the foreground.

## Related Issues

See #47 

## Remaining work
TODOs before this can be merged

- [x] Wait until #38 gets merged
- [ ] Merge master
- [ ] Test it

### CODE REVIEW

**Comment by genert** [comment]
This was largely based on @athornz solution. Please give it a look.

**Comment by genert** [comment]
Update: I will work on this tomorrow.

**Comment by Niek** [comment]
Any chance that this can be merged soon?

**Comment by Niek** [comment]
BTW, I'm getting this exception when trying to run the sample:
```bash
*** First throw call stack:
(
	0   CoreFoundation                      0x00007fff23b98bde __exceptionPreprocess + 350
	1   libobjc.A.dylib                     0x00007fff503b5b20 objc_exception_throw + 48
	2   CoreFoundation                      0x00007fff23bb9704 -[NSObject(NSObject) doesNotRecognizeSelector:] + 132
	3   CoreFoundation                      0x00007fff23b9d7bc ___forwarding___ + 1436
	4   CoreFoundation                      0x00007fff23b9f6c8 _CF_forwarding_prep_0 + 120
	5   Runner                              0x000000010153beea -[FLTFirebaseMessagingPlugin handleMethodCall:result:] + 858
	6   Flutter                             0x00000001031a84fd __45-[FlutterMethodChannel setMethodCallHandler:]_block_invoke + 104
	7   Flutter                             0x0000000103141ec0 _ZNK7flutter21PlatformMessage<…>
Exited (sigterm)
```
(my myBackgroundMessageHandler is static, so that shouldn't be causing it)

**Comment by genert** [comment]
Haven't had anytime to work on this. We ended up rolling native part for handling Firebase messaging due business logic requirements therefore we dropped using this library.

Perhaps, someone can take over or create a new one.

File: packages/firebase_messaging/README.md [hunk 1]
```diff
@@ -55,6 +55,90 @@ Note: When you are debugging on Android, use a device or AVD with Google Play se
...
+      public void registerWith(PluginRegistry registry) {
+        GeneratedPluginRegistrant.registerWith(registry);
+      }
+    }
+    ```
+1. Set name property of application in `AndroidManifest.xml`
+    ```
+    <application android:name=".Application" ...>
+    ```
+1. For iOS (Swift)
```
Comment (mudi256) [inline packages/firebase_messaging/README.md]: where to add this code in ios? like android we create an Application class

**Comment by athornz** [comment]
So is this PR dead?

**Comment by genert** [comment]
@athornz Yes, we went with native solution instead (APNs and notification service extension).

Will close this now, feel free to work on this.

**Pull request closed without merging**